### ISSUE TITLE
FPING for large target list

### ISSUE THREAD

**Issue by gregkuhnert** [phpservermon/phpservermon#757]
Monitoring is not scalable for a large number of targets at frequent polling intervals. I've got my cron job firing every minute. But if some hosts are down, the cron job will run again before the previous iteration is finished.

I've seen other tools using fping as an alternative. This tool is able to ping a large list of hosts in parallel. The existing cron job could be modified to get the list of ICMP hosts out to a file, and ping them all and store the output. Then, the existing cron job could be modified to parse the output from that file, and process as normal.

Would you consider this as a possible feature / enhancement in a future version?

**Comment by red-erik** [comment]
Hello,
according to my understanding, PHP Server Mon is a tool with a great flexibility to monitor services, tcp ports and so on and giving historical availability information. If your need is to monitor ICMP responses from a large number of host (and you are not interested in historical information) I suggest you some other interesting stuff like WhoisUp ([http://whoisup.sourceforge.net/](url). I used it a lot of times and the NMAP executable (required) can scan hundreds of host in less than 1 second. I only had to modify it a little bit to suite my desired configuration.
Anyway, the use of NMAP to monitor ICMP responses could be a good proposal for PhpServerMon too ;-)

Regards.

Red.

**Comment by MACscr** [comment]
Yep, I really like the core of the app and it keeps things simple, but definitely doesn't scale well unfortunately. Would love to see some options for that.
